src/EmailAccountRecoveryNew.sol

```diff
@@ -177,9 +181,26 @@ abstract contract EmailAccountRecoveryNew {
                 computeAcceptanceTemplateId(idx), acceptanceSubjectTemplates[account][idx]
             );
         }
-        for (uint256 idx = 0; idx < recoverySubjectTemplates[account].length; idx++) {
+
+        // TODO: can we get the validator module address from the emailAuthMsg before processing the
+        // entire subject template?
+        // address validatorModule; // TODO
+
+        string[][] memory recoverySubjectTemplate;
+        // get recovery subject template from the RecoverySubjectRegistry
+        if (account == validatorModule) {
```

Comment:
this would handle the case of the native safe account changeOwner recovery

Reply:
In order to minimise changes to this file, I'm wondering if this check can be done another way or in another contract.